uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -2335,4 +2335,292 @@ public function testToilToAccrueAmountIsSavedCorrectlyWhenAmountToAccrueIsAFloat
     sort($toilAmounts);
     $this->assertEquals($toilAmounts, $toilToAccrueAmounts);
   }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException
+   * @expectedExceptionMessage The TOIL amount plus all approved TOIL for current period is greater than the maximum for this Absence Type
+   */
+  public function testUpdatingToilThrowsExceptionWhenUpdatingApprovedToilWithAToilAmountGreaterThanWhatWasInitiallyApprovedAndTotalToilToBeAccruedIsGreaterThanMaximumForTheAbsenceType() {
```

Comment:
Woah, first prize for longest method name. I know you mentioned before about long method name, but is there any limit? :grin: